compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -734,6 +738,15 @@ private boolean isTupleTypeAssignable(BType source, BType target, List<TypePair>
 
     private boolean isTupleTypeAssignableToArrayType(BTupleType source, BArrayType target,
                                                      List<TypePair> unresolvedTypes) {
+
+        if (source.restType != null && target.state != BArrayState.UNSEALED) {
+            return false;
+        }
+
+        if (target.state != BArrayState.UNSEALED && source.tupleTypes.size() != target.size) {
```

Comment:
Shall we change L742-L748 to do the `target.state != BArrayState.UNSEALED` check once?

```ballerina
if (target.state != BArrayState.UNSEALED && (source.restType != null || source.tupleTypes.size() != target.size)) {
    return false;
}
```

Reply:
Fixed in dbeae05